src/components/TradeAssetSearch/components/GroupedAssetList/components/GroupedAssetRow.tsx

```diff
@@ -91,14 +91,19 @@ export const GroupedAssetRow = ({
         mx={2}
         _focus={focus}
       >
-        <Flex gap={4} alignItems='center'>
+        <Flex
+          gap={4}
+          alignItems='center'
+          maxWidth={
+            (isConnected || isDemoWallet) && !hideAssetBalance ? 'calc(100% - 100px)' : '100%'
```

Comment:
Do we need those checks?

Reply:
Yes to adapt the size depending if we show the balance or not